bd-proto-util/src/serialization/tests/macro_test.rs

```diff
@@ -0,0 +1,503 @@
+// shared-core - bitdrift's common client/server libraries
+// Copyright Bitdrift, Inc. All rights reserved.
+//
+// Use of this source code is governed by a source available license that can be found in the
+// LICENSE file or at:
+// https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt
+
+// We need to alias crate to bd_proto_util for the macro to work within the crate itself
+use crate as bd_proto_util;
+use anyhow::Result;
+use bd_macros::proto_serializable;
+use bd_proto_util::serialization::{ProtoFieldDeserialize, ProtoFieldSerialize, ProtoMessage};
+use protobuf::{CodedInputStream, CodedOutputStream, Message};
+
+#[test]
+fn test_simple_struct() -> Result<()> {
+  #[proto_serializable]
+  #[derive(Debug, PartialEq, Default)]
+  struct Foo {
+    a: u32,
+    b: String,
```

Comment:
FWIW I don't think we should support implicit field numbering. This seems like a footgun waiting to happen. I would probably remove it and make field numbers explicit and required. Also per offline convo in an optimal world we would check this against the .proto definitions. Fine as a TODO.